### ISSUE TITLE
Update tycho-compiler-jdt version

### ISSUE THREAD

**Issue by chris922** [mapstruct/mapstruct#1553]
During development of #1549 I noticed that there is a new version 1.2.0 for the `tycho-compiler-jdt` dependency, because the version that MapStruct uses right now (0.26.0) has a bug which prevents running the integration tests with the jdt_java8 compiler.

Upgrading to 1.2.0 wasn't done with just changing the version number. For a few other dependencies I had to stick the version number, because the `maven-enforcer-plugin` complained mixed version numbers.
Nevertheless some test cases were broken afterwards. It looked like issues with generics (`GenericsHierarchyTest` failed), maybe some more issues are there as well.

MapStruct should update to the newest version. Afterwards also the test for #1549 could be enabled for jdt_java8.

**Comment by chris922** [comment]
As tycho-compiler-jdt 1.3.0 is available I tried it once again (see referenced commit) and the test is still failing. 

For me it looks like a bug in JDT. I tried to find the cause and saw the `sourceReference` in `PropertyMappingBuilder` is of type `KeyOfAllBeings` but it should be an `AnimalKey`.

Maybe we should file a bug to the JDT team. A possible reproducer should have the same (strange) class hierarchy like we've got here:

```java
class AbstractAnimal
    implements Identifiable<KeyOfAllBeings>, IAnimal<AnimalKey> {
    // ...
}

interface Identifiable<T extends Key> {
    T getKey();
}

interface IAnimal<ID extends KeyOfAllBeings> extends Identifiable<KeyOfAllBeings> 
    @Override
    ID getKey();

    // ...
}
```

So the `IAnimal` overwrites the `getKey()` with the type `AnimalKey`. It is additionally last interface in the list of implementing interfaces of `AbstractAnimal` and thus methods defined here should overwrite the ones from `Identifiable`. 
It looks like JDT isn't considering this and thus during annotation processing the `getKey()` returns a `KeyOfAllBeings` instead of an `AnimalKey`

**Comment by filiphr** [comment]
Thanks for taking a deeper look @chris922. I also did that and found out the same. See https://github.com/mapstruct/mapstruct/pull/1587. I've also opened and issue for JDT https://bugs.eclipse.org/bugs/show_bug.cgi?id=540101 with a test case. Unfortunately, I haven't had the time to try and fix it in JDT myself

**Comment by chris922** [comment]
Ah.. good! I had in my mind that I read somewhere that you also checked this, but wasn't able to find it .. just looked at the issues and not the PRs :upside_down_face: 

Good to know that we both found the same cause independently! Nevertheless I had to add `ecj` to the list of libraries for the JUnit processor.. couldn't see something like this in your PR. Probably this wasn't required for version tycho-compiler-jdt 1.2.0?

In case you don't have a different issue we should link your PR to this issue.

**Comment by chris922** [comment]
At least you could now mention in the JDT ticket that the bug is also available in version 1.3.0 :smile:
